Common++/header/MacAddress.h

```diff
@@ -127,18 +143,47 @@ namespace pcpp
 		/// Allocates a byte array of length 6 and copies address value into it. Array deallocation is user
 		/// responsibility
 		/// @param[in] arr A pointer to where array will be allocated
+		/// @deprecated Use copyToNewBuffer instead as it returns a unique pointer to the allocated array.
+		PCPP_DEPRECATED("Use copyToNewBuffer instead as it returns a unique pointer to the allocated array.")
 		void copyTo(uint8_t** arr) const
 		{
-			*arr = new uint8_t[m_Address.size()];
-			std::copy(m_Address.begin(), m_Address.end(), *arr);
+			size_t size;
+			*arr = copyToNewBuffer(size).release();
 		}
 
 		/// Gets a pointer to an already allocated byte array and copies the address value to it.
 		/// This method assumes array allocated size is at least 6 (the size of a MAC address)
 		/// @param[in] arr A pointer to the array which address will be copied to
+		/// @remarks This method assumes that the provided array is at least 6 bytes long.
+		/// Prefer using the copyTo(uint8_t* buffer, size_t size) method if the array length is not guaranteed to be 6
+		/// bytes.
 		void copyTo(uint8_t arr[6]) const
 		{
-			std::copy(m_Address.begin(), m_Address.end(), arr);
+			return copyTo(arr, 6);
+		}
+
+		/// @brief Copies the address value to a user-provided buffer.
+		/// @param[in] buffer A pointer to the buffer where the address will be copied
+		/// @param[in] size The size of the buffer in bytes
+		/// @throws std::out_of_range If the provided size is smaller than 6 bytes.
+		void copyTo(uint8_t* buffer, size_t size) const
+		{
+			if (size < m_Address.size())
```

Comment:
I think we should compare if the size is 6?(then maybe we don't need to compare it?) It doesn't make sense to me to provide either a bigger or a smaller size.